### ISSUE TITLE
Unable to run GC, or update GC settings

### ISSUE THREAD

**Issue by agill17** [goharbor/harbor#13732]
What can we help you?

I am running harbor 2.1.1 on k8s ( backed by RDS, Elasticache and S3 ) and everytime I try to update GC settings ( for example schedule ) from the UI, I get Internal server error.

Here is what I see on core logs:
```
[harbor-harbor-core-64d9f99c48-w797h] 2020-12-10T18:17:01Z [ERROR] [/lib/http/error.go:54]: {"errors":[{"code":"INTERNAL_SERVER_ERROR","message":"job not found"}]}
```

Here is what I see on job-service logs:
```
[harbor-harbor-jobservice-6ff976878f-kmgzv] 2020-12-10T18:19:47Z [ERROR] [/jobservice/api/handler.go:288]: Serve http request 'POST /api/v1/jobs/fb1dadcece0255e2f8ae7078' error: 404 {"code":10010,"message":"object is not found","details":"fb1dadcece0255e2f8ae7078"}
```

Any pointers on how to fix this? I am unable to run GC at all.

**Comment by steven-zou** [comment]
This issue will be fixed at 2.1.2 that will be released later. For now, you can follow the guide shown below to fix it:

1. Follow the `step4-step6` shown [here](https://github.com/goharbor/harbor/wiki/Harbor-FAQs#stuck-in-read-only-mode) to make sure the GC schedule is removed
2. Reset DB data

```shell
docker exec -it harbor-db /bin/bash
psql -U postgres
\c registry
delete from admin_job where job_name = ‘IMAGE_GC’;
```
3. Then reset the schedule

**Comment by steven-zou** [comment]
@agill17 

could you help to check if the Redis pod in your cluster has been restarted?

**Comment by agill17** [comment]
2.1.2 was released 2 days ago, I am guessing it includes this fix? @steven-zou 
And I not running redis in a pod, its current on AWS Elasticache service.

**Comment by steven-zou** [comment]
@agill17 


This issue has been fixed in the 2.2 release and we'll also introduce a fix into the 2.1.5 patch release.

**Comment by steven-zou** [comment]
Can be fixed by pr 

- 1. https://github.com/goharbor/harbor/pull/14601
- 2. https://github.com/goharbor/harbor/pull/14597
